pkg/service/leases.go

```diff
@@ -104,47 +104,71 @@ func (s *LeasesService) ListLeases(acctID, principalID, nextAcctID, nextPrincipa
 	log.Infoln(string(jsonPayload))
 }
 
-func (s *LeasesService) LoginToLease(leaseID, loginProfile string, loginOpenBrowser, loginPrintCreds bool) {
+type leaseCreds struct {
+	AccessKeyID string `json:"accessKeyId,omitempty"`
+	ConsoleURL string `json:"consoleUrl,omitempty"`
+	ExpiresOn float64 `json:"expiresOn,omitempty"`
+	SecretAccessKey string `json:"secretAccessKey,omitempty"`
+	SessionToken string `json:"sessionToken,omitempty"`
+}
+
+func (s *LeasesService) Login(opts *LeaseLoginOptions) {
+	log.Debugln("Requesting leased account credentials")
+
+	params := &operations.PostLeasesAuthParams{}
+	params.SetTimeout(20 * time.Second)
+	res, err := apiClient.PostLeasesAuth(params, nil)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	responsePayload := res.GetPayload()
+
+	creds := leaseCreds(*responsePayload)
+	s.loginWithCreds(&creds, opts)
+}
+
+func (s *LeasesService) LoginByID(leaseID string, opts *LeaseLoginOptions) {
 	log.Debugln("Requesting leased account credentials")
 	params := &operations.PostLeasesIDAuthParams{
-		ID: leaseID,
+		ID:         leaseID,
 	}
-	params.SetTimeout(5 * time.Second)
+	params.SetTimeout(20 * time.Second)
```

Comment:
5 seconds seems a little short for a timeout.

In the future, this would be a nice thing to store in a global flag (or in the yaml file)